### PR TITLE
Add i.MX8Qm/QXP CAAM HAL

### PR DESCRIPTION
**Pull request by clementfaure**

### CODE REVIEW

File: core/drivers/crypto/caam/hal/imx_6_7/hal_cfg.c [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2020-2021 NXP
+ */
+#include <caam_hal_cfg.h>
+
+void caam_hal_cfg_setup_nsjobring(struct caam_jrcfg *jrcfg)
+{
+	caam_hal_cfg_common_setup_nsjobring(jrcfg);
```
Comment (jforissier) [inline core/drivers/crypto/caam/hal/imx_6_7/hal_cfg.c:9]: Wouldn't it make more sense to define `caam_hal_cfg_setup_nsjobring()` as `__weak` instead? And same for `caam_hal_rng_instantiated()`.

**Review by jforissier — COMMENTED** [review]
- One comment below on "drivers: caam: hal: add common initialization functions".
- For commits "drivers: caam: hal: add the support for imx8q" and "drivers: caam: add CAAM registers for imx8q platforms":
`Acked-by: Jerome Forissier <jerome@forissier.org>`

File: core/drivers/crypto/caam/hal/imx_6_7/hal_cfg.c (hunk 1 above)
Comment (clementfaure) [reply core/drivers/crypto/caam/hal/imx_6_7/hal_cfg.c:9]: Good suggestion.

**Comment by jforissier** [comment]
For "drivers: caam: hal: add common initialization functions", th ecode looks good, much simpler :) please don't forget to rewrite the commit subject accordingly. Also `core/drivers/crypto/caam/include/caam_hal_cfg.h` is not touched so no need for a copyright update.
With that:
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

**Comment by clementfaure** [comment]
Tags updated!

**Pull request merged**